### PR TITLE
Check for existence of release image before building ECK release

### PR DESCRIPTION
**Pull request by artemnikitin**
In the past we had situations when Jenkins update clear all history and it triggers release job to re-release Docker image.
This PR attempt to fix such behaviour. It adds small tool  written in Go to check for release image existence and add usage of this tool to Makefile and Jenkins release pipeline.
close https://github.com/elastic/cloud-on-k8s/issues/1104

### CODE REVIEW

**Comment by artemnikitin** [comment]
jenkins test this please

**Comment by artemnikitin** [comment]
jenkins test this please

**Comment by thbkrkr** [comment]
I'm a little bit concerned about this solution to implement the check of the existence of a docker image tag. The technical debt seems a little big for the service rendered:
- `github.com/heroku/docker-registry-client` does not seem well maintained ; can/should we use https://github.com/docker/distribution/tree/master/registry/client?
- we add another golang dependency to fetch to build or run this go program
- we add another Docker image to build ; I think this could be avoid that simply running the go program (compilation takes ~1s) `docker run --rm -v ... -w ... golang:1.12.5 go run main.go`

Since it will only be executing for release, it may not be a real problem.

https://github.com/genuinetools/reg might be another solution. We could download and install the binary in the ci Docker image (it weighs 12Mo). But I'm not sure that we can say it's much better maintained.

```bash
> curl -fSL "https://github.com/genuinetools/reg/releases/download/v0.16.0/reg-linux-amd64" -o "/usr/local/bin/reg"
> reg tags docker.elastic.co/eck/eck-operator | grep -c "^0.8.1$"
1                                                                                                                  
> reg tags docker.elastic.co/eck/eck-operator | grep -c "^0.9.0$"
0
```

More thoughts on this? cc @sebgl @pebrc

**Comment by thbkrkr** [comment]
There is also the solution to try to pull the image:

```bash
> docker pull docker.elastic.co/elasticsearch/elasticsearch:6.4.2 &>/dev/null; echo $?
0
                                                                                                                                                                                      
> docker pull docker.elastic.co/elasticsearch/elasticsearch:8.0.0 &>/dev/null; echo $?
1
```

**Comment by artemnikitin** [comment]
@thbkrkr I simplified everything as you suggested, thanks!

File: build/ci/release/Jenkinsfile [hunk 1]
```diff
@@ -63,3 +72,10 @@ pipeline {
     }
 
 }
+
+def releaseImageNotExist() {
+    return sh (
+        script: "docker pull $LATEST_RELEASED_IMG &>/dev/null; echo \$?",
+        returnStatus: true
```
Comment (thbkrkr) [inline build/ci/release/Jenkinsfile]: Not sure we need `; echo \$?` given we set `returnStatus: true`.

Comment (artemnikitin) [reply build/ci/release/Jenkinsfile]: Fixed

**Pull request merged**